File: Data Structures & Algorithms/is-palindrome/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isPalindrome(s) {
        let first = 0,last = s.length-1;
        const alpha = new Set("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789");
        while(first < last) {
            if(!alpha.has(s[first])) {
                first++
            } else if(!alpha.has(s[last])) {
                last--
            } else {
                if(!(s[first].toLowerCase()==s[last].toLowerCase())) {
                    return false;
                }
                first++
                last--
            }
        }
        return true
    }
}

File: Data Structures & Algorithms/search-insert-position/submission-0.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        first = 0
        last = len(nums)-1

        while first <= last:
            mid = first + (last-first) // 2
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                last = mid - 1
            else:
                first = mid + 1
                

        return first